va/va_test.go

```diff
@@ -235,16 +235,14 @@ func TestHTTP(t *testing.T) {
 	// TODO(#1989): close hs
 	hs := httpSrv(t, chall.Token)
 
-	goodPort, err := getPort(hs)
-	test.AssertNotError(t, err, "failed to get test server port")
+	va, goodPort, _, log := setup(t, hs)
```

Comment:
Since this is the only place where we use the returned HTTP port from `setup`, it's probably not worth adding the extra return value. For instance, here all we care about is that the `va.httpPort` point at a port that is not open. We could achieve the same thing by picking a random high port, which would probably have about the same (high) odds of being closed as what we do here, picking the assigned port + 1. Note that this method is somewhat complicated by the fact that it tries to put both "fail" and "succeed" cases in one test case. It would probably be better to split out the "fail" case, since that would allow `setup()` to straightforwardly point the VA at the `httpSrv`.